blocks/bricks/recurrent.py

```diff
@@ -991,8 +993,9 @@ def get_dim(self, name):
         return transition.get_dim(name)
 
     @application
-    def initial_state(self, state_name, batch_size, *args, **kwargs):
-        state_name, level = self.split_suffix(state_name)
-        transition = self.transitions[level]
-        return transition.initial_state(state_name, batch_size,
-                                        *args, **kwargs)
+    def initial_states(self, batch_size, *args, **kwargs):
+        kwargs['as_list'] = True
```

Comment:
Very much necessary. Without it the wrapper of transition.initial_states returns a tensor which you can't add to results in the way I did